src/main/java/org/folio/circulation/support/ItemRepository.java

```diff
@@ -187,6 +191,36 @@ private ItemRepository(
     }
   }
 
+  private CompletableFuture<Result<Collection<Item>>> fetchLoanTypes(Result<Collection<Item>> result) {
+    if (!fetchLoanType) {
+      return completedFuture(result);
+    }
+
+    return result.after(items -> {
+      Map<Item, String> itemToLoanTypeIdMap = items.stream()
+        .collect(Collectors.toMap(identity(), Item::determineLoanTypeForItem));
+
+      Set<String> loanTypeIdsToFetch = itemToLoanTypeIdMap.values().stream()
+        .filter(StringUtils::isNoneBlank)
+        .collect(Collectors.toSet());
+
+      return new MultipleRecordFetcher<>(loanTypesClient, "loantypes", identity())
+        .findByIds(loanTypeIdsToFetch)
+        .thenApply(mapResult(records -> records.toMap(byId())))
+        .thenApply(flatMapResult(loanTypes -> matchLoanTypesToItems(itemToLoanTypeIdMap, loanTypes)));
+    });
```

Comment:
Could this be done using the MultipleRecords class instead of collections

Reply:
This method is a link in a chain of methods which all have same input and return types, so maybe it makes sense to try and keep them uniform?

See method `findByIndexNameAndQuery`.

Reply:
If  you are referring to `MultipleRecords.findByIdIndexAndQuery`, that returns `MultipleRecords` and `MultipleRecords` has a combine method?

Have I misunderstood?

Reply:
I was referrring to `ItemRepository.findByIndexNameAndQuery`.

Reply:
That way well have predated the use of MultipleRecords. If this change doesn't make sense I will resolve this comment and it can be done at a later date